lm/lm_image.py

```diff
@@ -0,0 +1,87 @@
+__all__ = ["LmImageJob"]
+
+import shutil
+
+from sisyphus import *
+from i6_core import rasr
+from i6_core import util
+
+Path = setup_path(__package__)
+
+
+class LmImageJob(rasr.RasrCommand, Job):
+    """
+    pre-compute LM image without generating global cache
+    """
+
+    def __init__(
+        self,
+        crp,
+        extra_config=None,
+        extra_post_config=None,
+        encoding="utf-8",
+        mem=2,
+    ):
+        kwargs = locals()
+        del kwargs["self"]
+
+        self.config, self.post_config = LmImageJob.create_config(**kwargs)
+        self.exe = self.select_exe(crp.lm_util_exe, "lm-util")
+
+        self.log_file = self.log_file_output_path("lm_image", crp, False)
+        self.out_image = self.output_path("lm.image", cached=True)
```

Comment:
The number of images and their location can vary. I'd use the code from recognition/advanced_tree_search.py here to detect the images correctly and create all of them. In addition to the image paths themselves it would be good to have an updated config here as output (i.e. the original lm config + a post config with the images set correctly).

Reply:
There used to be logic to do just that and then I argued against it. Here is the original argument from the other thread:

>This reads as if this job takes a LM-config and prepares images for all the LMs.
Would it not be smarter/more efficient/more modular if this job only ever did one image at a time:
>
>E.g. if you have a 3 LMs and want to do all possible combinations (1, 2, 3, 1+2, 1+3, 2+3, 1+2+3) , then there would be 7 different LmImageJobs (because the config differs) and each individual image would be written 3 times.
>
>So better to have the Recognition Job do the logic of determining the number and identity of the LMs and then create n separate single LmImageJobs

Feel free to reopen the discussion.

Reply:
OK, you have a point there. But then we should add an assert that `lm_config.type == "arpa"`. And fix the name below.